### PR TITLE
#35: Refactor in TypeScript! (closes #35)

### LINKED ISSUE #35: Refactor in TypeScript!
## requirements
Refactor in TypeScript!

## specs
Refactor in TypeScript!

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Closes #35

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.}

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

File: src/InputChildren.tsx [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 
-export const Props = {
-  children: t.maybe(t.ReactChildren),
-  wrapper: t.maybe(t.Object)
-};
+export namespace InputChildren {
+  export type Props = React.InputHTMLAttributes<HTMLInputElement> & {
+    children?: any,
```
Comment (veej) [inline src/InputChildren.tsx]: `any` children? Can't be more precise?

File: styleguide/index.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
```
Comment (veej) [inline styleguide/index.html:1]: uhm... are we using tabs instead of spaces here? 🙅

**Review by veej — CHANGES_REQUESTED** [review]
few comments inline

File: src/InputChildren.tsx (hunk 1 above)
Comment (FrancescoCioria) [reply src/InputChildren.tsx]: can it? I thought TS didn't check this type anyway but only if it's undefined or not

File: styleguide/index.html (hunk 2 above)
Comment (FrancescoCioria) [reply styleguide/index.html:1]: :S

File: src/InputChildren.tsx (hunk 1 above)
Comment (veej) [reply src/InputChildren.tsx]: mmm, I think we used `React.ReactNode` or similar everywhere in `react-components`.
Some sort of check should happen somehow (see Tablo for example)

Comment (FrancescoCioria) [reply src/InputChildren.tsx]: done

**Review by veej — APPROVED** [review]
LGTM 👍

**Pull request merged**